### ISSUE TITLE
Improve testing code so that it's more robust on model updating 

### ISSUE THREAD

**Issue by nanjiangshu** [MetabolicAtlas/MetabolicAtlas#1005]
**Please describe the feature**

As a developer, I want the testing code to be more robust so that it's more robust when models are updated.

**Acceptance Criteria**

- [ ] Broken tests due to model updating are fixed
- [ ] Tests are improved so that they are more robust on model updating

**Additional context**

Related PR https://github.com/MetabolicAtlas/data-files/pull/29

**Comment by MalinAhlberg** [comment]
Potentially good to know for this issue: the current versions are available in `api/test/jest.setup.js` (parsed from `api/src/data/integratedModels.json`)

>

**Comment by e0** [comment]
Please keep in mind that while it would be less tedious to have more robust testing code, it's important to do this in a way that it's not automated to the point where it doesn't detect potential problems. For example if we know for a fact that a model has 1000 reactions and we introduce a bug in data generation that creates 2000 reactions, how should we set the truth for what the number of reactions ought to be?

**Comment by nanjiangshu** [comment]
> Please keep in mind that while it would be less tedious to have more robust testing code, it's important to do this in a way that it's not automated to the point where it doesn't detect potential problems. For example if we know for a fact that a model has 1000 reactions and we introduce a bug in data generation that creates 2000 reactions, how should we set the truth for what the number of reactions ought to be?

I agree. We can not relax all kinds of testing conditions for the sake of robustness.  There is a balance to keep.

**Comment by perjo** [comment]
> For example if we know for a fact that a model has 1000 reactions and we introduce a bug in data generation that creates 2000 reactions, how should we set the truth for what the number of reactions ought to be? 

This should be covered by tests in the data-generation project. Issse link to be added @pdmnyberg

**Comment by pdmnyberg** [comment]
Issue in `data-generation` https://github.com/MetabolicAtlas/data-generation/issues/37
